File: tests/fixtures/textmate/zig/basic.zig
const std = @import("std");

const Reading = struct {
    label: []const u8,
    value: f64,
};

/// Return null rather than divide an empty slice; Unicode: café 🚀 𝌆.
fn average(values: []const f64) ?f64 {
    if (values.len == 0) return null;
    var total: f64 = 0;
    for (values) |value| {
        total += value;
    }
    return total / @as(f64, @floatFromInt(values.len));
}

pub fn main() !void {
    const readings = [_]Reading{
        .{ .label = "café 🚀", .value = 0x1.5p+3 },
        .{ .label = "orbit 𝌆", .value = 2.5e1 },
    };
    const values = [_]f64{ readings[0].value, readings[1].value };
    const mean = average(&values) orelse return error.EmptyReadings;
    std.debug.print("{s}: {d:.2}\n", .{ readings[0].label, mean });
}

File: tests/fixtures/textmate/rust/basic.rs
#![allow(dead_code)]

/// One borrowed sensor label, including café, rockets 🚀, and 𝌆.
#[derive(Debug, Clone, PartialEq)]
struct Reading<'a> {
    label: &'a str,
    value: f64,
}

fn summarize(readings: &[Reading<'_>]) -> Result<String, &'static str> {
    let first = readings.first().ok_or("no readings")?;
    let peak = readings.iter().map(|item| item.value).reduce(f64::max).unwrap();
    let raw = r#"units: "km/s" 🚀 𝌆"#;
    Ok(format!("{:?}: peak {peak:.1} {raw}", first.label))
}

fn main() {
    let readings = [Reading {
        label: "café",
        value: 0x2A as f64,
    }];
    match summarize(&readings) {
        Ok(message) => println!("{message}"),
        Err(error) => eprintln!("error: {error}"),
    }
}
